lib/matplotlib/tests/test_dates.py

```diff
@@ -511,6 +511,162 @@ def _create_auto_date_locator(date1, date2):
         assert strings == expected
 
 
+def test_concise_formatter_formats():
+    formats = ['%Y', '%m/%Y', 'day: %d',
+               '%H hr %M min', '%H hr %M min', '%S.%f sec']
+
+    def _create_auto_date_locator(date1, date2):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(locator, formats=formats)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = []
+        for st in ax.get_yticklabels():
+            sts += [st.get_text()]
+        return sts
+
+    d1 = datetime.datetime(1997, 1, 1)
+    results = ([datetime.timedelta(weeks=52 * 200),
+    [str(t) for t in range(1980,
+    2201,
+    20)]],
+    [datetime.timedelta(weeks=52),
+    ['1997',
+    '02/1997',
+    '03/1997',
+    '04/1997',
+    '05/1997',
+    '06/1997',
+    '07/1997',
+    '08/1997',
+    '09/1997',
+    '10/1997',
+    '11/1997',
+    '12/1997']],
+    [datetime.timedelta(days=141),
+    ['01/1997',
+    'day: 22',
+    '02/1997',
+    'day: 22',
+    '03/1997',
+    'day: 22',
+    '04/1997',
```

Comment:
I don't know that this should be laid out with so many carriage returns.  I appreciate it may make it a bit easier to debug, but it makes it harder for other to scroll through the file.